### ISSUE TITLE
Question about additionalProperties

### ISSUE THREAD

**Issue by yangcen6** [allegro/allegro-api#10075]
`{
    "header": {
        "code": 200,
        "desc": "success"
    },
    "body": {
        "id": "71ebbb5d-515b-4b46-9921-8c314bba54b5",
        "deliveryMethodId": "0aafb43c-e66a-46ec-9cc4-29bb39ebb483",
        "sender": {
            "name": "PLSwieckoDC",
            "company": "XIAOMI",
            "street": "Poland",
            "streetNumber": "850396639",
            "postalCode": "69-100",
            "city": "Poland",
            "countryCode": "PL",
            "email": "[allegro@xiaomi.com](mailto:allegro@xiaomi.com)",
            "phone": "791444014",
            "point": "AL015WE1"
        },
        "receiver": {
            "name": "SapulakJoanna",
            "company": "",
            "street": "Warszawa",
            "streetNumber": "856906331",
            "postalCode": "02-677",
            "city": "Warszawa",
            "countryCode": "PL",
            "email": "[djdrl48ng7+2e3906584@allegromail.pl](mailto:djdrl48ng7+2e3906584@allegromail.pl)",
            "phone": "+48602194438",
            "point": "AL015WE1"
        },
        "packages": [
            {
                "waybill": "A001329J66",
                "type": "PACKAGE",
                "length": {
                    "value": 18,
                    "unit": "CENTIMETER"
                },
                "width": {
                    "value": 9,
                    "unit": "CENTIMETER"
                },
                "height": {
                    "value": 13,
                    "unit": "CENTIMETER"
                },
                "weight": {
                    "value": 0.946,
                    "unit": "KILOGRAMS"
                }
            }
        ],
        "createdDate": "2024-09-20T08:27:47.539Z",
        "carrier": "ALLEGRO",
        "labelFormat": "PDF",
        "additionalServices": [

        ],
        "additionalProperties": {
            "EXTERNAL_CARRIER_WAYBILL": "1Z83567X6829384178",
            "FIRST_MILE_CARRIER": "UPS"
        },
        "traceId": "68e4eff6eded7867",
        "httpCode": "200"
    }
}`
LIke this order a0b4fe50-8670-11ef-95de-87deba44aa82, is the field additionalProperties-EXTERNAL_CARRIER_WAYBILL requried? If not, when will you push this to us.

**Comment by AureliuszBrussy** [comment]
You can check what additional properties exist for a particular delivery method by using [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation#operation/getDeliveryServices). There, we return whether a field is read-only, whether you can pass it, and whether it is required (if you can pass it). 

In this case, two carriers handle this delivery method, so in additionalProperties we return the second waybill that can appear on the label.

**Comment by yangcen6** [comment]
What do you mean  two carriers handle this delivery method?

**Comment by AureliuszBrussy** [comment]
This means that the so-called first mile can be carried out by one courier company, and the delivery, or so-called last mile, by another company. In this case, a particular parcel may have two waybills.

**Comment by yangcen6** [comment]
However, the logistics number on the shipping label we received is the logistics number of the second courier company, not the logistics number of the first courier company.
![img_v2_f04f5c4d-a173-414a-a9a2-78d75c55d71l](https://github.com/user-attachments/assets/960332f1-af98-4604-b9eb-be2f9e45ffcd)

**Comment by AureliuszBrussy** [comment]
Basically, it looks like this: the package is picked up by UPS, but it is delivered to the Allegro parcel machine. UPS has its waybill, Allegro has its waybill (you can also see it on label). You [can read more about this delivery method in Allegro's help](https://help.allegro.com/sell/pl/a/jak-wysylac-z-allegro-one-punkt-box-b2lmAnaaxtx). If you have additional questions about the delivery methods themselves, please use the [contact form](https://allegro.pl/pomoc/kontakt).

**Comment by yangcen6** [comment]
Is this bar code related to the tracking number of the ups? And the two carriers method is only for Allegro OneBox?

**Comment by AureliuszBrussy** [comment]
Yes, EXTERNAL_CARRIER_WAYBILL is related to UPS barcode. No, it is not only for Allegro One Box. As I wrote before, you can check it by using [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation#operation/getDeliveryServices). For all methods with this case in the `additionalProperties` array will return `EXTERNAL_CARRIER_WAYBILL` entity.

**Comment by AureliuszBrussy** [comment]
You can check tracking by using [GET /order/carriers/{carrierId}/tracking](https://developer.allegro.pl/documentation#operation/getParcelTrackingUsingGET). Use the waybill from `waybill` field.

**Comment by yangcen6** [comment]
For this case order  a0b4fe50-8670-11ef-95de-87deba44aa82, which waybill number do we need to use to get the real delivered status, A001329J66 or 1Z83567X6829384178?

**Comment by AureliuszBrussy** [comment]
As I wrote, use the waybill returned in the `waybill` field for simplification. In this case, it will be A001329J66. But the latter will also work. You must then query UPS for the status.

**Comment by yangcen6** [comment]
Why can't we get the delivery information of this order 23fa1640-8961-11ef-9806-0b2fec43d54a through this express tracking number 1003138594456U?
`{"header":{"code":200,"desc":"success"},"body":{"carrierId":"ALLEGRO","waybills":[{"waybill":"1003138594456U"}],"traceId":"934996615094851c","httpCode":"200"}}`

**Comment by AureliuszBrussy** [comment]
If you want to check tracking by using `EXTERNAL_CARRIER_WAYBILL`, you should use `FIRST_MILE_CARRIER` `id` in the `carrierId` path parameter, in this case - DPD (so path should look like this: **/order/carriers/DPD/tracking**).

**Comment by yangcen6** [comment]
What is the FIRST_MILE_CARRIER id in this case？

**Comment by AureliuszBrussy** [comment]
The same case as with UPS we talk before.

**Comment by yangcen6** [comment]
According to your suggestion, we changed the carrierId to DPD, but still cannot find the delivery information of this order.
request：
`{"path_param":{"carrierId":"DPD"},"query_param":{"waybill":"1003138594456U"},"shop_id":"123550430"}`
respond：
`{"header":{"code":200,"desc":"success"},"body":{"carrierId":"DPD","waybills":[{"waybill":"1003138594456U"}],"traceId":"2472cf3780dd6f88","httpCode":"200"}}`

**Comment by AureliuszBrussy** [comment]
I have consulted the issue with the team responsible for tracking and I apologize for the mistake. In such cases, you should simply use the master waybill, as it is the one assigned to the Allegro order. If you were sending a DPD shipment under WzA (and not Allegro One), you could track the DPD shipment successfully. 

Allegro One package has its status history collected from several systems - DPD is only a subcontractor and its job is to get the package to the machine. In this case, the DPD waybill would end up as “delivered”, and in fact the package can still waiting to be picked up by the buyer.

**Comment by yangcen6** [comment]
OK，thanks
